### PR TITLE
Converted decision task timeout to optional

### PR DESCRIPTION
**Pull request by mfateev**
**What changed?**
Converted decision task timeout from required to optional. The default is 10 seconds overridable per namespace through dynamic config.

**Why?**
Decision task timeout is a non trivial concept to grasp and rarely requires non default value.
Making it optional will improve onboarding experience.

**How did you test it?**
I added a unit test and ran all the unit tests locally.

**Potential risks**
The decision task timeout will be incorrectly set.

### CODE REVIEW

File: common/service/dynamicconfig/constants.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ var keys = map[Key]string{
 	EnableNamespaceNotActiveAutoForwarding: "system.enableNamespaceNotActiveAutoForwarding",
 	TransactionSizeLimit:                   "system.transactionSizeLimit",
 	MinRetentionDays:                       "system.minRetentionDays",
-	MaxDecisionStartToCloseSeconds:         "system.maxDecisionStartToCloseSeconds",
+	MaxDecisionTaskStartToCloseTimeout:     "system.maxDecisionTaskStartToCloseTimeout",
```
Comment (alexshtin) [inline common/service/dynamicconfig/constants.go:71]: Timeout is just timeout: it should not have `Max` prefix.

File: common/service/dynamicconfig/constants.go [hunk 2]
```diff
@@ -297,8 +298,8 @@ const (
 	TransactionSizeLimit
 	// MinRetentionDays is the minimal allowed retention days for namespace
 	MinRetentionDays
-	// MaxDecisionStartToCloseSeconds is the minimal allowed decision start to close timeout in seconds
-	MaxDecisionStartToCloseSeconds
+	// MaxDecisionTaskStartToCloseTimeout is the minimal allowed decision start to close timeout in seconds
+	MaxDecisionTaskStartToCloseTimeout
```
Comment (alexshtin) [inline common/service/dynamicconfig/constants.go:302]: Same here.

File: service/frontend/workflowHandler.go [hunk 3]
```diff
@@ -342,7 +342,7 @@ func (wh *WorkflowHandler) StartWorkflowExecution(ctx context.Context, request *
 		return nil, wh.error(errInvalidExecutionStartToCloseTimeoutSeconds, scope)
 	}
 
-	if request.GetTaskStartToCloseTimeoutSeconds() <= 0 {
+	if request.GetTaskStartToCloseTimeoutSeconds() < 0 {
```
Comment (alexshtin) [inline service/frontend/workflowHandler.go:345]: Why is this? Does `0` mean: don't wait a single nanosecond?

File: common/service/dynamicconfig/constants.go (hunk 1 above)
Comment (mfateev) [reply common/service/dynamicconfig/constants.go:71]: This is not a timeout. This is the maximum value allowed. If user specifies a higher one it is going to be lowered to the configured maximum.

File: service/frontend/workflowHandler.go (hunk 3 above)
Comment (mfateev) [reply service/frontend/workflowHandler.go:345]: 0 mean "use default"

Comment (mfateev) [reply service/frontend/workflowHandler.go:345]: Or we need to switch all of them to some nullable type

Comment (alexshtin) [reply service/frontend/workflowHandler.go:345]: I would leave it as `0`.

File: common/service/dynamicconfig/constants.go (hunk 1 above)
Comment (alexshtin) [reply common/service/dynamicconfig/constants.go:71]: Sounds little weird but makes sense now.

Comment (mfateev) [reply common/service/dynamicconfig/constants.go:71]: Usually setting this timeout to a high value is a mistake. That's why we limit it.

**Pull request merged**